src/FakeItEasy/Expressions/ParsedCallExpression.cs

```diff
@@ -16,7 +16,7 @@ internal class ParsedCallExpression
         {
             this.CalledMethod = calledMethod;
             this.ArgumentsExpressions = argumentsExpressions;
-            this.callTarget = new Lazy<object>(() => callTargetExpression?.Evaluate());
+            this.callTarget = new Lazy<object>(() => callTargetExpression?.Evaluate() !);

```

Comment:
Two things:
- Why the `?` (I know, it was already there)?. `callTargetExpression` is of type `Expression` (not nullable), so it should never be null. And if it is, it's an error.
- How do we know `Evaluate` won't return null?

Reply:
You're right, `callTargetExpression` could evaluate to `null`. In particular when clients try to configure a static method. We catch the nullness/staticness later, but for a while, we can have a null call target.
I've fixed up. The nullability propagated bit. In 4 places I resorted to the `!` operator. In 3 of them, it's fairly obvious that it's safe. In `PropertyExpressionHelper`, I added comments to clear it up a bit…

Longer term, we might want to consider refactoring the determination of whether a method can be intercepted, but I'm leery of mucking about too much in the middle of the nullability transformation...

Reply:
> I'm leery of mucking about too much in the middle of the nullability transformation...

Totally agree!